### DISCUSSION TITLE
GraphQL with Laravel: HasMany Relationship Loading Incorrect Data, Works Fine with REST API

### DISCUSSION THREAD

**Discussion by Atif-Javeed** [api-platform/api-platform#2795] [Q&A] [answered]
I'm using Laravel with API Platform and working with a HasMany relationship. When I retrieve data using GraphQL, it returns incorrect results for the related entities, even though the same query works perfectly when I use the REST API.
 
Has anyone faced a similar issue, or are there specific configurations needed for GraphQL to properly handle HasMany relationships in Laravel API Platform?

**Comment by soyuka** [comment]
@toitzi any idea? did you try this?

**Reply by toitzi** [reply]
Yeah can confirm, i will look into it.

**Reply by toitzi** [reply]
After digging a bit deeper, the issue basicly is:

With `HasMany` everything in that table gets loaded. It does not matter where it belongs to, the information on that seems to get lost on the way. for example:

You have a `users` table with 2 entries (`id` = `1` and `id` = `2`) and a `user_phones` table with 2 entries (`user_id` = `1` and `user_id` = 2). For both users both `user_phones` table entries are loaded and returned.

Im still in the process of digging deeper but the issue seems to be inside [here](https://github.com/api-platform/core/blob/main/src/Laravel/Eloquent/State/LinksHandler.php)

Where eloquent links only seem to be handled for http operations (REST)
![CleanShot 2024-11-12 at 23 09 08@2x](https://github.com/user-attachments/assets/11b30e6a-55a3-4550-81a5-d991be3c04b4)

Have not figured out a solution yet, still wrapping my head arround what's happening.

**Reply by toitzi** [reply]
A hint on what is happening might be inside the `ReadProvider` or graphql

![CleanShot 2024-11-12 at 23 52 32@2x](https://github.com/user-attachments/assets/1f1b9b68-63f1-43c5-bd16-3d6accf72164)


There seems to be a different approach in doctrine using `linkClass` and `linkProperty`, both of which are defined there but unused inside laravel graphql.

**Comment by shakeel-nodesol** [comment]
> I'm using Laravel with API Platform and working with a HasMany relationship. When I retrieve data using GraphQL, it returns incorrect results for the related entities, even though the same query works perfectly when I use the REST API.
> 
> Has anyone faced a similar issue, or are there specific configurations needed for GraphQL to properly handle HasMany relationships in Laravel API Platform?

I also face the same issue.

**Comment by toitzi** [comment, marked as answer]
I opened a draft PR

https://github.com/api-platform/core/pull/6792

which aims to fix the problem.

**Reply by shakeel-nodesol** [reply]
Thanks, It's working now
